freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java

```diff
@@ -64,4 +111,114 @@ public void deleteHost(HostRequest request) {
         //TODO  Will be implemented CDPSDX-515
         return;
     }
+
+    private Stack getFreeIpaStack(String envCrn) {
+        FreeIpaClient freeIpaClient = null;
+        String failure;
+        String accountId = crnService.getCurrentAccountId();
+        LOGGER.debug("Looking for stack using env:{} and account-id:{}", envCrn, accountId);
+        return stackService.getByEnvironmentCrnAndAccountId(envCrn, accountId);
+    }
+
+    private String getRealm(Stack stack) {
```

Comment:
I think it would make sense to throw this here: `throw new KeytabCreationException(EMPTY_REALM);` and you don't have to inspect ion the caller side if it's null.